Physics3D/geometry/triangleMesh.cpp

```diff
@@ -177,31 +181,82 @@ TriangleMesh::TriangleMesh(UniqueAlignedPointer<float>&& vertices, UniqueAligned
 TriangleMesh::TriangleMesh(int vertexCount, int triangleCount, const Vec3f* vertices, const Triangle* triangles) :
 	MeshPrototype(vertexCount, triangleCount) {
 
-	size_t vertexOffset = getOffset(vertexCount);
+        size_t vertexOffset = 8;//getOffset(vertexCount);
 
+	
+	//printf("vertex_count = ori:%d, now: %lu\ntriangle_count = ori:%d, now: %lu\n", vertexCount, getOffset(vertexCount) * 3, triangleCount, getOffset(triangleCount) * 3),
+	
+	//printf("TriangleMesh::TriangleMesh->vertex_offset = %lu\n", vertexOffset);
+	
+	//printf("TriangleMesh::TriangleMesh->vertexCount = %d\n", vertexCount);
+	
 	float* xValues = this->vertices.get();
 	float* yValues = xValues + vertexOffset;
 	float* zValues = yValues + vertexOffset;
 
+	unsigned int a = 0;
 	for(size_t i = 0; i < vertexCount; i++) {
-		xValues[i] = vertices[i].x;
-		yValues[i] = vertices[i].y;
-		zValues[i] = vertices[i].z;
+	    xValues[a] = vertices[i].x;
+	    yValues[a] = vertices[i].y;
+	    zValues[a] = vertices[i].z;
+	    /*  printf("x = %f\n", xValues[a]);
+	    printf("y = %f\n", yValues[a]);
+	    printf("z = %f\n", zValues[a]);
+	    */a++;
+	    a+=vertexOffset*((a%vertexOffset)==0)*2;
+	    //printf("a = %u\n", a);
+	    
 	}
-	fixFinalBlock(this->vertices.get(), vertexCount);
 
-	size_t triangleOffset = getOffset(triangleCount);
+	unsigned int last_index = a-1;
+	unsigned int size_left =  (((vertexCount + 7) & 0xFFFFFFFFFFFFFFF8)*3-vertexCount*3)/3;
+	//	printf("a*3 = %d", a*3);
+	//	printf("size_left = %u\n", size_left);
+	for(unsigned int i=a; i<a+size_left; i++){
+	  xValues[i] = xValues[last_index];
+	  yValues[i] = yValues[last_index];
+	  zValues[i] = zValues[last_index];
+	  //  printf("xValues[%d] = %f\n", i, xValues[i]);
+	}
+	
+	//fixFinalBlock(this->vertices.get(), vertexCount);
 
+	size_t triangleOffset = 8;//getOffset(triangleCount);
+
+	
+	
 	int* aValues = this->triangles.get();
 	int* bValues = aValues + triangleOffset;
 	int* cValues = bValues + triangleOffset;
 
+	a = 0;
 	for(size_t i = 0; i < triangleCount; i++) {
-		aValues[i] = triangles[i].firstIndex;
-		bValues[i] = triangles[i].secondIndex;
-		cValues[i] = triangles[i].thirdIndex;
+        	aValues[a] = triangles[i].firstIndex;
+		bValues[a] = triangles[i].secondIndex;
+		cValues[a] = triangles[i].thirdIndex;
+		/*	printf("a = %d\n", aValues[a]);
+		printf("b = %d\n", bValues[a]);
+		printf("c = %d\n", cValues[a]);
+		*/a++;
+		a+=triangleOffset*((a%triangleOffset)==0)*2;
+	        //printf("a_index = %u\n", a);
 	}
-	fixFinalBlock(this->triangles.get(), triangleCount);
+
+	
+	last_index = a-1;
+	size_left =  ((((triangleCount + 7) & 0xFFFFFFFFFFFFFFF8)*3-triangleCount*3)/3);
```

Comment:
Don't reuse the same variable. Create a new variable. This code could actually be extracted into a static function